### PR TITLE
Added deployment of templates from workbooks

### PR DESCRIPTION
**Pull request by stankis**

### CODE REVIEW

**Comment by robot-charts** [comment]
**i18n-check**
<details>
<summary>The following components have not been translated</summary>
<p><a href="https://weblate.gravity-ui.com/projects/datalens/charts-9085-deploy-workbook-templates-607/component-collections-structure-607/en/">component.collections-structure (en)</a><br><a href="https://weblate.gravity-ui.com/projects/datalens/charts-9085-deploy-workbook-templates-607/component-collections-structure-607/ru/">component.collections-structure (ru)</a></p>
</details>

Wait for the reviewers to check your changes in Weblate and try running github action again.

**Comment by robot-charts** [comment]
**i18n-check**
Please merge the Pull Request with the changes from Weblate into your branch.

**Comment by robot-charts** [comment]
**i18n-check**
Please merge the Pull Request with the changes from Weblate into your branch.

**Comment by robot-charts** [comment]
**i18n-check**
<details>
<summary>The following components have not been translated</summary>
<p><a href="https://weblate.gravity-ui.com/projects/datalens/charts-9085-deploy-workbook-templates-607/component-dialog-relations-view-607/en/">component.dialog-relations.view (en)</a><br><a href="https://weblate.gravity-ui.com/projects/datalens/charts-9085-deploy-workbook-templates-607/component-dialog-relations-view-607/ru/">component.dialog-relations.view (ru)</a></p>
</details>

Wait for the reviewers to check your changes in Weblate and try running github action again.

**Comment by robot-charts** [comment]
**i18n-check**
<details>
<summary>The following components have not been translated</summary>
<p><a href="https://weblate.gravity-ui.com/projects/datalens/charts-9085-deploy-workbook-templates-607/component-dialog-relations-view-607/en/">component.dialog-relations.view (en)</a><br><a href="https://weblate.gravity-ui.com/projects/datalens/charts-9085-deploy-workbook-templates-607/component-dialog-relations-view-607/ru/">component.dialog-relations.view (ru)</a></p>
</details>

Wait for the reviewers to check your changes in Weblate and try running github action again.

**Review by imsitnikov — CHANGES_REQUESTED** [review]

File: src/ui/components/CollectionsStructure/DeployEntryToWorkbookDialog.tsx [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+                    onClose: () => {
+                        dispatch(closeDialog());
+                    },
+                },
+            }),
+        );
+    };
+};
+
+export const DeployEntryToWorkbookDialog: React.FC<Props> = ({open, onApply, onClose}) => {
```
[the remark this answers is not in the document]
Comment (stankis) [reply src/ui/components/CollectionsStructure/DeployEntryToWorkbookDialog.tsx:47]: done

File: src/ui/components/CollectionsStructure/DeployEntryToWorkbookDialog.tsx [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+    const handleMigrateEntry = React.useCallback(
```
[the remark this answers is not in the document]
Comment (stankis) [reply src/ui/components/CollectionsStructure/DeployEntryToWorkbookDialog.tsx:48]: of course, fixed

File: src/ui/units/connections/store/actions/index.ts [hunk 3]
```diff
@@ -261,13 +264,15 @@ export function createConnection(name: string, dirPath?: string) {
...
         }
 
-        batch(() => {
-            if (templateFolderId) {
-                history.replace(`/navigation/${templateFolderId}`);
-            } else if (connectionId) {
-                history.replace(`/connections/${connectionId}`);
-            }
+        if (workbookId) {
+            history.replace(`/workbooks/${workbookId}?tab=all`);
```
[the remark this answers is not in the document]
Comment (stankis) [reply src/ui/units/connections/store/actions/index.ts:268]: let's do if it is already fixed in main

File: src/ui/components/CollectionsStructure/DeployEntryToWorkbookDialog.tsx [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+    );
+
+    const isLoadingMigrateEntry = useSelector(selectIsLoadingMigrateEntriesToWorkbook);
+
+    return (
+        <CollectionStructureDialog
+            open={open}
+            type={ResourceType.Workbook}
+            initialCollectionId={null}
+            canSelectInitialCollectionId={false}
```
[the remark this answers is not in the document]
Comment (stankis) [reply src/ui/components/CollectionsStructure/DeployEntryToWorkbookDialog.tsx:70]: fixed

File: src/ui/components/CollectionsStructure/DeployEntryToWorkbookDialog.tsx [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+            targetTitle?: string;
+        }) => {
+            if (targetWorkbookId) {
+                await onApply(targetWorkbookId);
+            }
+        },
+        [onApply],
+    );
+
+    const isLoadingMigrateEntry = useSelector(selectIsLoadingMigrateEntriesToWorkbook);
```
[the remark this answers is not in the document]
Comment (stankis) [reply src/ui/components/CollectionsStructure/DeployEntryToWorkbookDialog.tsx:63]: yes, it is not implemented at all ) copy-pasted from Migration component and forgot to implement for deployment

**Comment by robot-charts** [comment]
**i18n-check**
<details>
<summary>The following components have not been translated</summary>
<p><a href="https://weblate.gravity-ui.com/projects/datalens/charts-9085-deploy-workbook-templates-607/component-dialog-relations-view-607/en/">component.dialog-relations.view (en)</a><br><a href="https://weblate.gravity-ui.com/projects/datalens/charts-9085-deploy-workbook-templates-607/component-dialog-relations-view-607/ru/">component.dialog-relations.view (ru)</a></p>
</details>

Wait for the reviewers to check your changes in Weblate and try running github action again.

**Comment by robot-charts** [comment]
**i18n-check**
<details>
<summary>The following components have not been translated</summary>
<p><a href="https://weblate.gravity-ui.com/projects/datalens/charts-9085-deploy-workbook-templates-607/component-dialog-relations-view-607/en/">component.dialog-relations.view (en)</a><br><a href="https://weblate.gravity-ui.com/projects/datalens/charts-9085-deploy-workbook-templates-607/component-dialog-relations-view-607/ru/">component.dialog-relations.view (ru)</a><br><a href="https://weblate.gravity-ui.com/projects/datalens/charts-9085-deploy-workbook-templates-607/component-dialog-switch-public-view-607/ru/">component.dialog-switch-public.view (ru)</a></p>
</details>

Wait for the reviewers to check your changes in Weblate and try running github action again.

**Comment by robot-charts** [comment]
**i18n-check**
<details>
<summary>The following components have not been translated</summary>
<p><a href="https://weblate.gravity-ui.com/projects/datalens/charts-9085-deploy-workbook-templates-607/component-dialog-relations-view-607/en/">component.dialog-relations.view (en)</a><br><a href="https://weblate.gravity-ui.com/projects/datalens/charts-9085-deploy-workbook-templates-607/component-dialog-relations-view-607/ru/">component.dialog-relations.view (ru)</a><br><a href="https://weblate.gravity-ui.com/projects/datalens/charts-9085-deploy-workbook-templates-607/component-dialog-switch-public-view-607/ru/">component.dialog-switch-public.view (ru)</a></p>
</details>

Wait for the reviewers to check your changes in Weblate and try running github action again.

**Comment by robot-charts** [comment]
**i18n-check**
<details>
<summary>The following components have not been translated</summary>
<p><a href="https://weblate.gravity-ui.com/projects/datalens/charts-9085-deploy-workbook-templates-607/component-dialog-relations-view-607/en/">component.dialog-relations.view (en)</a><br><a href="https://weblate.gravity-ui.com/projects/datalens/charts-9085-deploy-workbook-templates-607/component-dialog-relations-view-607/ru/">component.dialog-relations.view (ru)</a><br><a href="https://weblate.gravity-ui.com/projects/datalens/charts-9085-deploy-workbook-templates-607/component-dialog-switch-public-view-607/ru/">component.dialog-switch-public.view (ru)</a></p>
</details>

Wait for the reviewers to check your changes in Weblate and try running github action again.

**Comment by robot-charts** [comment]
**i18n-check**
<details>
<summary>The following components have not been translated</summary>
<p><a href="https://weblate.gravity-ui.com/projects/datalens/charts-9085-deploy-workbook-templates-607/component-dialog-relations-view-607/en/">component.dialog-relations.view (en)</a><br><a href="https://weblate.gravity-ui.com/projects/datalens/charts-9085-deploy-workbook-templates-607/component-dialog-relations-view-607/ru/">component.dialog-relations.view (ru)</a><br><a href="https://weblate.gravity-ui.com/projects/datalens/charts-9085-deploy-workbook-templates-607/component-dialog-switch-public-view-607/ru/">component.dialog-switch-public.view (ru)</a></p>
</details>

Wait for the reviewers to check your changes in Weblate and try running github action again.

File: src/ui/components/CollectionsStructure/DeployTemplateToWorkbookDialog.tsx [hunk 6]
```diff
@@ -0,0 +1,99 @@
...
+            await dispatch(
+                copyTemplate({
+                    templateName,
+                    workbookId,
+                    productId,
+                }),
+            );
+            onApply(workbookId);
+        },
+        [dispatch, templateName, onApply],
```
Comment (imsitnikov) [inline src/ui/components/CollectionsStructure/DeployTemplateToWorkbookDialog.tsx:78]: `productId`

**Review by korvin89 — APPROVED** [review]

File: src/ui/units/connections/store/actions/index.ts [hunk 7]
```diff
@@ -261,13 +264,17 @@ export function createConnection(name: string, dirPath?: string) {
             flow([resetFormsData, dispatch])();
         }
 
-        batch(() => {
-            if (templateFolderId) {
+        if (templateFolderId) {
```
Comment (imsitnikov) [inline src/ui/units/connections/store/actions/index.ts:267]: This will not work with workbook templates

**Review by imsitnikov — CHANGES_REQUESTED** [review]

File: src/ui/units/connections/store/actions/index.ts (hunk 7 above)
Comment (stankis) [reply src/ui/units/connections/store/actions/index.ts:272]: agree, fixed

**Comment by robot-charts** [comment]
**i18n-check**
<details>
<summary>The following components have not been translated</summary>
<p><a href="https://weblate.gravity-ui.com/projects/datalens/charts-9085-deploy-workbook-templates-607/new-workbooks-607/ru/">new-workbooks (ru)</a></p>
</details>

Wait for the reviewers to check your changes in Weblate and try running github action again.

**Comment by robot-charts** [comment]
**i18n-check**
<details>
<summary>The following components have not been translated</summary>
<p><a href="https://weblate.gravity-ui.com/projects/datalens/charts-9085-deploy-workbook-templates-607/new-workbooks-607/ru/">new-workbooks (ru)</a></p>
</details>

Wait for the reviewers to check your changes in Weblate and try running github action again.

**Review by imsitnikov — APPROVED** [review]

**Pull request merged**